indico/modules/events/editing/service.py

```diff
@@ -123,3 +124,27 @@ def service_get_status(event):
     except requests.RequestException as exc:
         return {'status': None, 'error': unicode(ServiceRequestFailed(exc))}
     return {'status': resp.json(), 'error': None}
+
+
+def service_handle_editable(editable):
+    revision = editable.revisions[-1]
+    data = {
+        'files': EditingRevisionFileSchema().dump(revision.files, many=True),
+        'endpoints': {
+            'revisions': {
+                'replace': url_for('.api_replace_revision', revision, _external=True)
+            },
+            'file_upload': url_for('.api_upload', editable.contribution, type=editable.type.name, _external=True)
```

Comment:
You can just pass the `editable` to get everything since is locator includes the contribution's locator and the type.